meta/SaiSerialize.cpp

```
@@ -5440,3 +5500,79 @@ void sai_deserialize_redis_link_event_damping_aied_config(
     sai_deserialize_number(j["decay_half_life"], value.decay_half_life, false);
     sai_deserialize_number(j["flap_penalty"], value.flap_penalty, false);
 }
+
+/** @brief deserializes the stats capability list 
+*
+*   Iterates thru stat_enum_str and populate
+*   the stats_capability with respective
+*   stat_enum (String to Enum conversion).
+*   Also iterates thru stat_modes_str and populate
+*   the stats_capability with respective
+*   stat_modes (String to Enum conversion)
+* 
+*   @param stats_capability stats capability enum list
+*   @param stat_enum_str SAI stat enum list as string
+*   @param stat_modes_str SAI stat mode list as string
+*   @return Void
+*/
+void sai_deserialize_stats_capability_list(
+        _Inout_ sai_stat_capability_list_t *stats_capability,
+        _In_    const std::string& stat_enum_str,
+        _In_    const std::string& stat_modes_str)
+{
+    SWSS_LOG_ENTER();
+
+    if (stats_capability == NULL)
+    {
+        SWSS_LOG_THROW("Stats capability pointer in deserialize is NULL");
+    }
+
+    uint32_t num_capabilities = stats_capability->count;
+    size_t stat_enum_position = 0;
+    size_t stat_modes_position = 0;
+
+    for (uint32_t i = 0; i < num_capabilities; i++)
+    {
+        /* 1. Populate stat_enum */
+        size_t old_stat_enum_position = stat_enum_position;
+        stat_enum_position = stat_enum_str.find(",", old_stat_enum_position);
+        std::string stat_enum = stat_enum_str.substr(old_stat_enum_position,
+                stat_enum_position - old_stat_enum_position);
+        stats_capability->list[i].stat_enum = std::stoi(stat_enum);
+
+        /* We have run out of values to add to our list */
+        if (stat_enum_position == std::string::npos)
+        {
+            if (num_capabilities != i + 1)
+            {
+                SWSS_LOG_WARN("Lesser stat_enums than expected: expected %d, received %d",
+                        num_capabilities, i+1);
+            }
```

this should throw as data is invalid

---

Addressed the comment.